js/utils.js

```diff
@@ -415,6 +415,162 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
+        return {
+          today: moment().add('days', smartParseFloat(options.modifierValue)).startOf('day'),
+          inputDate: isEntryContainsTime
+            ? moment(options.date, 'HH.mm').startOf('day')
+            : moment(options.date).startOf('day')
+        };
+
+      case 'todayaddmonths':
+        return {
+          today: moment().add('month', smartParseFloat(options.modifierValue)).startOf('day'),
+          inputDate: isEntryContainsTime
+            ? moment(options.date, 'HH.mm').startOf('day')
+            : moment(options.date).startOf('day')
+        };
+
+      case 'todayaddyears':
+        return {
+          today: moment().add('year', smartParseFloat(options.modifierValue)).startOf('day'),
+          inputDate: isEntryContainsTime
+            ? moment(options.date, 'HH.mm').startOf('day')
+            : moment(options.date).startOf('day')
+        };
+
+      case 'nowsubtractminutes':
+        return {
+          today: getDate().subtract('minute', smartParseFloat(options.modifierValue)).startOf('minute'),
+          inputDate: isEntryContainsTime
+            ? getDate(options.date, 'HH.mm')
+            : getDate(options.date)
+        };
+
+      case 'nowsubtracthours':
+        return {
+          today: getDate().subtract('hour', smartParseFloat(options.modifierValue)).startOf('minute'),
+          inputDate: isEntryContainsTime
+            ? getDate(options.date, 'HH.mm')
+            : getDate(options.date)
+        };
+
+      case 'todayminusdays':
+        return {
+          today: moment().subtract('days', smartParseFloat(options.modifierValue)).startOf('day'),
+          inputDate: isEntryContainsTime
+            ? moment(options.date, 'HH.mm').startOf('day')
+            : moment(options.date).startOf('day')
+        };
+
+      case 'todayminusmonths':
+        return {
+          today: moment().subtract('month', smartParseFloat(options.modifierValue)).startOf('day'),
+          inputDate: isEntryContainsTime
+            ? moment(options.date, 'HH.mm').startOf('day')
+            : moment(options.date).startOf('day')
+        };
+
+      case 'todayminusyears':
+        return {
+          today: moment().subtract('year', smartParseFloat(options.modifierValue)).startOf('day'),
+          inputDate: isEntryContainsTime
+            ? moment(options.date, 'HH.mm').startOf('day')
+            : moment(options.date).startOf('day')
+        };
+
+      default:
+        break;
+    }
+  }
+
+
+  function isDateMatches(options) {
+    if (options && options.filterModifier) {
+      var result = getDateModifiedValues({
+        date: options.date,
+        filterModifier: options.filterModifier.default.value,
+        modifierValue: options.filterModifier.default.offset,
+        isDeviceTimeZoneEnabled: options.filterModifier.default.isDeviceTimeZoneEnabled
+      });
+
+      switch (options.condition) {
+        case 'dateis':
+          return moment(result.inputDate).isSame(result.today);
```

Comment:
It looks like `result.today` is calculated statically, i.e. does not depend on the value of each record, but is being creating from scratch every single time in the `getDateModifiedValues()` function, e.g. in `case 'now'`, this value is `today: getDate().startOf('minute')`. Why does this need to be created every single time instead of using a cached value?

Reply:
I cannot move this variable to another place because in this case we can have a scenario in between comparison where we will have in From value with using time zone and in To value without using time zone

Reply:
Assuming there is a filter that uses date comparison, this is my breakdown:

1. `isDateMatches()` is getting called as many times as the the number of records
2. `getDateModifiedValues()` is getting called as many times as there are the number of records
3. The `getDateModifiedValues()` runs something like `getDate().add('minute', smartParseFloat(options.modifierValue)).startOf('minute')` as many times as the number of records. However, `getDate().add('minute', smartParseFloat(options.modifierValue)).startOf('minute')` should produce the same output every single time. If the data source has 10,000 entries, we don't need to compute this value 10,000 times.

You need to fix this problem.

I suggest making sure that the value you pass to `.isSame()`, `.isBefore()` etc. only gets calculated only once for each filter.